test/integration/sequelize-custom.test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const { expect } = require('chai');
+const Sequelize = require('../../index');
+const config = require('../config/config');
+
+const dialectName = process.env.DIALECT === 'postgres-native' ? 'postgres' : process.env.DIALECT;
+const CustomDialect = require(`../../lib/dialects/${dialectName}`);
+
+before(() => {
+  const options = {
+    ...config[dialectName],
+    dialect: CustomDialect
+  };
+  
+  this.sequelize = new Sequelize(options);
+});
+
+after(() => {
+  this.sequelize.close();
+});
+
+describe('Custom Dialect', () => {
+  describe('authenticate', () => {
+    it('connects to database successfully', async () => {
+      try {
+        await this.sequelize.authenticate();
+        expect(true).to.equal(true);
+      } catch (error) {
+        console.error(error);
+      }
```

Comment:
This is not a safe way to test promises. If this promise `rejects`, the test suite will not fail, just log. The safe way to do it is:

```suggestion
      return expect(this.sequelize.authenticate()).resolves.to.equal(true);
```